lease-kubernetes/src/main/scala/akka/coordination/lease/kubernetes/internal/KubernetesApiImpl.scala

```diff
@@ -227,7 +227,7 @@ PUTs must contain resourceVersions. Response:
   }
 
   private def pathForLease(name: String): Uri.Path =
-    Uri.Path.Empty / "apis" / "akka.io" / "v1" / "namespaces" / namespace / "leases" / name
+    Uri.Path.Empty / "apis" / "akka.io" / "v1" / "namespaces" / namespace / "leases" / name.toLowerCase.filter(c => "\\d|\\w|-|\\.".r.findAllIn(c.toString).nonEmpty)
```

Comment:
Seems a bit 'heavy' to use a regex for this (and construct it for every single character), but this is probably not called often so might not be a problem.

I like the new behavior :).

Reply:
Yeah I couldn't think of another way to do it easily without having a list of characters either allowed or not allowed. Perhaps a list of allowed is better?

Reply:
maybe something like `name.toLowerCase.filter(c => c.isLetterOrDigit || c == '-' || c == '.')`?

Reply:
I do it better tomorrow. I left my computer now. I also noticed the test failure was due to upping version of Akka Http and now it wants us to specify the content type. I managed to get the test to pass before I left . I push tomorrow.

Reply:
Great, looking forward to it!

Reply:
After various performance tests, the one I just pushed was by far the most performant. 

with 10K iteration 
0.77 seconds for the new one 
vs
16.61 sec for your one (which, by the way was more performant that my initial attempt) 

![Screenshot from 2020-11-26 09-51-55](https://user-images.githubusercontent.com/637276/100298923-507b6f80-2fcd-11eb-813f-058593ddbffb.png)